float64[] position

bool is_reachable
